scan.go

```diff
@@ -221,12 +242,29 @@ func (s *ScanService) Do() (*ScanCursor, error) {
 		path += "/" + strings.Join(typesPart, ",")
 	}
 
+	// Define the body of the request
+	var body map[string]interface{}
+
+	if s.source != nil {
+		body = s.source
+	} else {
+		src, err := s.searchSource.Source()
+
+		if err != nil {
+			return nil, err
+		}
+
+		// searchSource.Source() actually returns a map, yet its return type
+		// is specified as interface...
```

Comment:
`SearchSource.Source()` does not return a `map[string]interface{}`. [It returns an `interface{}`](https://github.com/olivere/elastic/blob/release-branch.v3/search_source.go#L295). You depend on the internals of that function, which can change. All that `Source()` promises is that you can serialize it to JSON, which you can do with a variety of types. The [`Query`](https://github.com/olivere/elastic/blob/release-branch.v3/query.go) establishes the same contract BTW.

Please provide an example of a function signature in Go that specifies the following contract: This function either returns `io.Reader`, `string`, `[]byte`, `*json.RawMessage`, or `map[string]interface{}`.